### PR TITLE
CORE-12881 Patch: Fix freeldr memory enumeration

### PR DESCRIPTION
**Pull request by tkreuzer**
See https://jira.reactos.org/browse/CORE-12881
Split in 2 commits

### CODE REVIEW

**Comment by HBelusca** [comment]
Looks good; do you think, @tkreuzer, that these ULONGLONG_ALIGN_UP etc... macros could also be useful for our x64 port of ROS (especially for the kernel / hal and drivers) ?

**Comment by tkreuzer** [comment]
@HBelusca Fox x64 we don't need it, since on _WIN64 ULONG_PTR == ULONGLONG, so there the official macro from wdm.h works just fine. It's only required for 64 bit values (like physical memory addresses) on a 32 bit system. There could be a few other places where it would be useful, like hal or some low level drivers, but I don't really know where. If we find more places, we can move it to some other header.

File: boot/freeldr/freeldr/arch/i386/pcmem.c [hunk 1]
```diff
@@ -311,7 +318,7 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
                       PcBiosMapCount,
                       PcBiosMemoryMap[PcBiosMapCount].BaseAddress,
                       PcBiosMemoryMap[PcBiosMapCount].Length);
-                continue;
+                goto nextRange;
```
Comment (ThFabba) [inline boot/freeldr/freeldr/arch/i386/pcmem.c]: This change seems like it also deserves to be separated

**Comment by jeditobe** [comment]
Serge has sent email to Timo with his information.

**Comment by learn-more** [comment]
@tkreuzer A heads-up, your author name is still set as 'tkreuzer'

**Comment by jeditobe** [comment]
Let's have this one committed?

**Comment by jeditobe** [comment]
Why do I still do not see it on code.reactos.org?

**Comment by gigaherz** [comment]
The code.reactos.org database is still indexing commit data -- Colin said it may take days.

**Pull request merged**